Gems/LmbrCentral/Code/Source/Builders/LevelBuilder/LevelBuilderWorker.cpp

```diff
@@ -155,35 +155,18 @@ namespace LevelBuilder
     {
         PopulateOptionalLevelDependencies(sourceRelativeFile, productPathDependencies);
 
-        AZStd::binary_semaphore extractionCompleteSemaphore;
-        auto extractResponseLambda = [&]([[maybe_unused]] bool success) {
-            AZStd::string levelsubfolder;
-            AzFramework::StringFunc::Path::Join(tempDirectory.c_str(), "level", levelsubfolder);
+        std::future<bool> extractResult;
+        AzToolsFramework::ArchiveCommandsBus::BroadcastResult(
+            extractResult, &AzToolsFramework::ArchiveCommandsBus::Events::ExtractArchive, levelPakFile, tempDirectory);
 
-            PopulateLevelSliceDependencies(levelsubfolder, productDependencies, productPathDependencies);
-            PopulateMissionDependencies(levelPakFile, levelsubfolder, productPathDependencies);
-            PopulateLevelAudioControlDependencies(levelPakFile, productPathDependencies);
+        [[maybe_unused]] bool result = extractResult.get();
```

Comment:
This should just call std::future<T>::wait

```suggestion
        extractResult.wait();
```

Reply:
Will do.